controllers/csi.go

```diff
@@ -0,0 +1,689 @@
+		Args:            args,
+		Resources:       resourceRequirements,
+		VolumeMounts:    volumeMounts,
+		ImagePullPolicy: corev1.PullIfNotPresent,
+		Env:             env,
+	}
+
+	return csiSnapshotter
+}
+
+func (s *sideCarContainer) getCephCsiContainer(pluginType, driverName string, controller bool) *corev1.Container {
+	containerName := ""
+	switch pluginType {
+	case "cephfs":
+		containerName = "csi-cephfsplugin"
+	case "rbd":
+		containerName = "csi-rbdplugin"
+	}
+	if containerName == "" {
+		return nil
+	}
+	// csi plugin container
+	args := []string{
+		"--nodeid=$(NODE_ID)",
+		"--endpoint=$(CSI_ENDPOINT)",
+		"--v=5",
+		"--pidlimit=-1",
+		fmt.Sprintf("--type=%s", pluginType),
+		fmt.Sprintf("--controllerserver=%t", controller),
+		fmt.Sprintf("--drivername=%s", driverName),
+	}
+
+	resourceRequirements := corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{
+			// Empty for now
+		},
+		Requests: corev1.ResourceList{
+			// Empty for now
+		},
+	}
+
+	volumeMounts := []corev1.VolumeMount{
+		{Name: "socket-dir", MountPath: defaultSockerDir},
+		{Name: "host-dev", MountPath: "/dev"},
+		{Name: "host-sys", MountPath: "/sys"},
+		{Name: "lib-modules", MountPath: "/lib/modules", ReadOnly: true},
+		{Name: "ceph-csi-configs", MountPath: "/etc/ceph-csi-config"},
+		{Name: "keys-tmp-dir", MountPath: "/tmp/csi/keys"},
+	}
+
+	if !controller {
+		biDirectionalMount := corev1.MountPropagationBidirectional
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{Name: "host-run-mount", MountPath: "/run/mount"})
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{Name: "csi-plugins-dir", MountPath: fmt.Sprintf("%s/plugins/", defaultKubeletDirPath), MountPropagation: &biDirectionalMount})
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{Name: "pods-mount-dir", MountPath: fmt.Sprintf("%s/pods", defaultKubeletDirPath), MountPropagation: &biDirectionalMount})
+	}
+	// encryption is only supported for rbd not for cephfs
+	if pluginType == "rbd" {
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{Name: "ceph-csi-kms-config", MountPath: "/etc/ceph-csi-encryption-kms-config/", ReadOnly: true})
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{Name: "oidc-token", MountPath: "/run/secrets/tokens", ReadOnly: true})
+	}
+
+	env := []corev1.EnvVar{
+		{
+			Name: "POD_IP",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "status.podIP",
+				},
+			},
+		},
+		{
+			Name: "NODE_ID",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "spec.nodeName",
+				},
+			},
+		},
+		{
+			Name: "POD_NAMESPACE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.namespace",
+				},
+			},
+		},
+		{
+			Name:  "CSI_ENDPOINT",
+			Value: defaultSocketPath,
+		},
+	}
+
+	cephCsiPlugin := &corev1.Container{
+		Name:            containerName,
```

Comment:
Since it's not used anywhere else, what about `containerName := fmt.Sprintf("csi-%splugin", pluginType)`?